Thematic commentary of the week

SteffenAstheimer opened the pull request:
Implements a carousel on the index page that switches between the word-of-the-day and a new thematicCommentary-of-the-week.

### To-Do:

**Display:**
- [x] Fix display for smaller screens:
   - [x] button placement
   - [x] adjust carousel item placement to screen size
   - [x] add padding for smaller screens with missing carousel images
   - [x] adjust font size for mobile devices
- [x] Fix word-of-the-day implementation
   - [x] check for problematic word-of-the-day lengths (so far none found)
   - [x] adjust font size for smaller screens
   - [x] ensure correct display if carousel is absent (it appears that with the new split into includes-templates the word-of-the-day is lacking a container/row if it is standalone)
- [x] align images and flipcard
   - [x] works for most sizes
   - [x] doesn't work if text is too long 
- [x] fix bug with flipcard "depth" -> should be the same effect as for portraits
- [x] Find/Create static image for the left side
- [x] Translate static teaser text bits
- [x] format italic text and quoted segments
- [x] Carousel height before loading content

**Functionality:**
- [x] Hide Carousel until birthday
- [x] Take image out of the thematic commentary (right side)

SteffenAstheimer committed:
create html and css framework for weekly thematic commentary

still missing copyright marks for the pictures;
known problems:
- front and back of flipcards don't align
- word of the day needs to dynamically adjust font size to the container size (also width doesn't align with thematic commentaries)
- rendering only works on medium/large screens -> account for mobile devices
- arrows and "hover zones" to change slides don't overlap / block each other out

SteffenAstheimer committed:
add function to retrieve commentary of the week...

... fix placement of carousel buttons and collision with word of the day
delete hover zones

peterstadler committed:
add new function wega-util:compute-image-url#2

peterstadler committed:
improve app:thematicCommentary-of-the-week#3

* support URL parameter "tc" for requesting a specific thematicCommentary independent of the date
* get teaser image

peterstadler committed:
move HTML fragments into includes folder

peterstadler committed:
improve app:thematicCommentary-of-the-week#3

* add comments
* make more robust regarding faulty data or multiple user inputs
* fix  'thematicCommentaryOfTheWeek-url'

SteffenAstheimer committed:
adjust positioning of carousel control buttons

and clean up artifacts from old hover zones

SteffenAstheimer committed:
adjust image sizing and placing

SteffenAstheimer committed:
lower carousel indicators

SteffenAstheimer committed:
improve text distribution on the flipcards

SteffenAstheimer committed:
improve wega-util:compute-image-url#2

add resolver for simple file names as url

SteffenAstheimer committed:
improve rendering of changing graphics

SteffenAstheimer committed:
improve alignment and and optical presentation

add margin to flipcard to align height with images
move fading effect of the images outwards to give them more space and improve the feeling of having one consistent carousel item
add highlights to the buttons and increase the clickable range to make them easy to spot

SteffenAstheimer committed:
adjust carousel for different screen sizes

works up to smallest screens, needs font size adaption for mobile devices though
also the padding appears to be lost on smaller screens

SteffenAstheimer committed:
fix structure of wordOfTheDay template

SteffenAstheimer committed:
refactor carousel to support AJAX loading and query params

peterstadler commented on modules/app.xqm:
```diff
@@ -564,8 +564,9 @@ declare
-        let $today := current-date()
+    %templates:default("otdDate", "")
+    function app:thematicCommentary-of-the-week($node as node(), $model as map(*), $lang as xs:string, $otdDate as xs:string) as map(*) {
+        let $today := if ($otdDate ne "" and $otdDate castable as xs:date) then xs:date($otdDate) else current-date()
```
`"" castable as xs:date` ergibt `false()`, daher brauchst Du eigentlich nicht explizit auf den leeren String prüfen.
```suggestion
        let $today := if ($otdDate castable as xs:date) then xs:date($otdDate) else current-date()
```

peterstadler reviewed (approved):
I just tested the functionality and it works as expected. You have noted some more improvements yourself that should be implemented before being deployed to production, but I think this is good to go for internal review.

peterstadler commented:
Here's an example of a too long wordOfTheDay (so the link at the bottom is missing):
<img width="1179" height="515" alt="Bildschirmfoto 2025-10-27 um 09 53 41" src="https://github.com/user-attachments/assets/66a27dce-cdd5-4678-91d3-2b36fe92ba2c" />

peterstadler commented:
While testing, I found the carousel images could be improved a little bit. Don't know how exactly, but maybe a bit more padding, or some frame, or a vignette clipping or effect?

SteffenAstheimer committed:
enable highlighting and quotation for teasers

SteffenAstheimer committed:
translate static teaser bits

SteffenAstheimer committed:
display single quotes in flipcards

SteffenAstheimer committed:
improve link placement on flipcards

SteffenAstheimer committed:
switch from css to xslt for text styling

SteffenAstheimer committed:
add dynamic font sizes to carousel headers

and delete carousel indicators (too much trouble with covering content on different screen sizes)

SteffenAstheimer committed:
keep padding on TC-of-the-week row for small screens

SteffenAstheimer committed:
move link out of button zone

SteffenAstheimer committed:
align images towards center

peterstadler committed:
rework ajax-loader

and make sure that the main page features a word-of-the-day even if the ajax load fails

Co-Authored-By: SteffenAstheimer <steffen.astheimer@uni-paderborn.de>

peterstadler committed:
remove `%templates:wrap` annotation

to get rid of extra div in the HTML output

SteffenAstheimer committed:
Set trigger to 'click' for flip-cards on mobile devices

SteffenAstheimer committed:
add jubilee image for teaser carousel...

move image fading to hover effect on carousel buttons (to prevent hiding image text)
add desaturation and sepia filters to images to balance 'popping' colors
add blur to transparent flip-card

peterstadler reviewed (approved):
tested locally and it looks great 👍
